web3.js/src/transaction.js

```diff
@@ -432,38 +444,41 @@ export class Transaction {
    * Verify signatures of a complete, signed Transaction
    */
   verifySignatures(): boolean {
-    return this._verifySignatures(this.serializeMessage());
+    return this._verifySignatures(this.serializeMessage(), true);
   }
 
   /**
    * @private
    */
-  _verifySignatures(signData: Buffer): boolean {
-    let verified = true;
+  _verifySignatures(signData: Buffer, requireAllSignatures: boolean): boolean {
     for (const {signature, publicKey} of this.signatures) {
+      if (signature === null) {
+        return !requireAllSignatures;
+      }
       if (
         !nacl.sign.detached.verify(signData, signature, publicKey.toBuffer())
       ) {
-        verified = false;
+        return false;
       }
     }
-    return verified;
+    return true;
   }
 
   /**
    * Serialize the Transaction in the wire format.
-   *
-   * The Transaction must have a valid `signature` before invoking this method
    */
-  serialize(): Buffer {
-    const {signatures} = this;
-    if (!signatures || signatures.length === 0) {
-      throw new Error('Transaction has not been signed');
-    }
+  serialize(config?: SerializeConfig): Buffer {
+    const {requireAllSignatures, verifySignatures} = Object.assign(
+      {requireAllSignatures: true, verifySignatures: true},
+      config,
+    );
 
     const signData = this.serializeMessage();
-    if (!this._verifySignatures(signData)) {
-      throw new Error('Transaction has not been signed correctly');
+    if (
+      verifySignatures &&
+      !this._verifySignatures(signData, requireAllSignatures)
+    ) {
+      throw new Error('Transaction has not been signed');
```

Comment:
nit: I miss the "correctly" in the error message because the transaction could be signed with a signature that fails verification

Reply:
fixed